src/utils/wallet.js

```diff
@@ -718,12 +722,15 @@ class Wallet {
     }
 
     async setupRecoveryMessage(accountId, method, securityCode, recoverySeedPhrase) {
-        const { secretKey } = parseSeedPhrase(recoverySeedPhrase)
-        const recoveryKeyPair = KeyPair.fromString(secretKey)
-        await this.validateSecurityCode(accountId, method, securityCode);
-        const newPublicKey = recoveryKeyPair.publicKey
-        await this.addNewAccessKeyToAccount(accountId, newPublicKey)
-        await store.dispatch(redirectTo('/profile', { globalAlertPreventClear: true }))
+        const { publicKey } = parseSeedPhrase(recoverySeedPhrase)
+        await this.validateSecurityCode(accountId, method, securityCode)
+        try {
+            await this.addNewAccessKeyToAccount(accountId, publicKey)
+        } catch(e) {
+            throw new WalletError(e, 'errors.recoveryMethods.setupMethod')
```

Comment:
@Patrick1904 I think expected first argument to `WalletError` is error message, not another error. Probably best to do `console.error` with caught error.

Reply:
@vgrichina fixed it here: https://github.com/near/near-wallet/pull/1128/commits/f11ee9270bd751a6ecfa8f856474cfd0ead588ff